src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java

```diff
@@ -212,4 +216,21 @@ public void deleteGcpCloudContext(UUID workspaceId, AuthenticatedUserRequest use
         .addParameter(WorkspaceFlightMapKeys.WORKSPACE_ID, workspaceId)
         .submitAndWait(null);
   }
+
+  /**
+   * Get the GCP project associated with a workspace. If none is defined, throw. This helper method
+   * is used by other classes that need to know the GCP project associated with the workspace.
+   *
+   * @param workspaceId unique workspace id
+   * @return GCP project id
+   */
+  public String getGcpProject(UUID workspaceId) {
```

Comment:
Why not make this return `Optional<String>` instead of throw? Then callers could throw whatever they want or handle it more quitely.

Reply:
I renamed it `getRequiredGcpProject`. I believe the common case is where we are retrieving the project in order to use it for something.